### PR TITLE
Enable .NET 5 SDK

### PR DESCRIPTION
**Pull request by stevejgordon**
Support using newer compiler features such as Source Generators.

### CODE REVIEW

**Comment by stevejgordon** [comment]
Investigating failures

File: benchmarks/Elastic.Transport.Benchmarks/Elastic.Transport.Benchmarks.csproj [hunk 1]
```diff
@@ -3,11 +3,12 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <IsPackable>false</IsPackable>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net5.0</TargetFramework>
+    <LangVersion>9.0</LangVersion>
```
Comment (stevejgordon) [inline benchmarks/Elastic.Transport.Benchmarks/Elastic.Transport.Benchmarks.csproj:7]: There is a bug in 0.12.1 which means that we cannot use latest here from `Directory.Build.props` so temporarily fixing this to 9.0.

Once Benchmark.NET includes a fix, we can remove this after updating to that version.

File: benchmarks/Elastic.Transport.Benchmarks/Elastic.Transport.Benchmarks.csproj [hunk 2]
```diff
@@ -3,11 +3,12 @@
...
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="BenchmarkDotNet" Version="0.12.0" />
+    <PackageReference Include="BenchmarkDotNet" Version="0.12.1" />
```
Comment (stevejgordon) [inline benchmarks/Elastic.Transport.Benchmarks/Elastic.Transport.Benchmarks.csproj:11]: Required for net5.0 TFM support.

File: benchmarks/Elastic.Transport.Benchmarks/Elastic.Transport.Benchmarks.csproj (hunk 1 above)
Comment (Mpdreamz) [reply benchmarks/Elastic.Transport.Benchmarks/Elastic.Transport.Benchmarks.csproj:7]: Is there already an open a ticket for this on Benchmark.NET ?

Comment (stevejgordon) [reply benchmarks/Elastic.Transport.Benchmarks/Elastic.Transport.Benchmarks.csproj:7]: Yep, it's fixed and will be in 0.12.2 - https://github.com/dotnet/BenchmarkDotNet/pull/1420

**Review by Mpdreamz — APPROVED** [review]
LGTM!

**Pull request merged**